components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -241,12 +241,16 @@ public Map<String, Integer> getActionsCountPerType(Integer tenantId) throws Acti
     }
 
     @Override
-    public Action getActionByActionId(String actionId, Integer tenantId) throws ActionMgtException {
+    public Action getActionByActionId(String actionType, String actionId, Integer tenantId) throws ActionMgtException {
 
         try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false)) {
             Action action = getActionBasicInfoById(dbConnection, actionId, tenantId);
```

Comment:
Shouldn't this action needs to be retrieved passing the action type ? There's no existence for an action without action type right ?

If so, ideally the error below should not happen

Reply:
As the cache is against the action Type, the method in the interface required to have action type as parameter (Both CacheBackedActionMgtDAO class and this class implements same interface). 

The action is retrieved from the database solely based on the action Id. So don't we need to check the action type of the retrieved action with action type used in API path. Eg: Otherwise user will able to retrieve pre-issue access token action by its Id, with "/authentication" in the API path.

Reply:
Update the SQL to get action based on both action id and action type.

Reply:
Addressed with bc8c0ad77e8ada99ee15df865b39682c1aaa3e58